gnomad_qc/v4/create_release/create_false_dup_liftover.py

```diff
@@ -0,0 +1,176 @@
+            )
+        )
+    )
+
+    return ht
+
+
+def main(args):
+    """
+    There are three clinically relevant genes impacted by false duplications in the GRCh38 reference (KCNE1, CBS, and CRYAA).
+    To make it easier for our users to switch to gnomAD v4 and the new reference build,
+    this script creates a new release file that combines information from the v2 exomes and genomes only for these three genes.
+    To create this file, filter the v2 liftover files (https://gnomad.broadinstitute.org/downloads#v2-liftover) to these genes,
+    and then merge frequency information across the exomes and genomes
+    """
+    # Read in liftover files.
+    exome_ht = filter_liftover_to_false_dups("exomes")
+    genome_ht = filter_liftover_to_false_dups("genomes")
+
+    # Union and merge to contain any variant present in either.
+    # Note: only about ~550 overlap, with ~187k exome variants and ~12k genome variants.
+    # This makes sense in exonic regions.
+    # Genome global annotations stored with _1
+    ht = exome_ht.select().join(genome_ht.select(), how="outer")
+
+    # Annotate with information from both exomes and genomes - many will be NaN.
+    ht = ht.annotate(
+        v2_exomes=exome_ht[ht.locus, ht.alleles],
+        v2_genomes=genome_ht[ht.locus, ht.alleles],
+    )
+
+    # Checkpoint output to new temp file.
+    ht = ht.checkpoint(hl.utils.new_temp_file("three_dup_genes_liftover", "ht"))
+
+    # Merge exomes and genomes frequencies.
+    # Code adapted from
+    # gnomad_qc/v4/create_release/create_combined_faf_release_ht.py#L155
+    # as of 01-26-2025
+    # Changed for 'gen_anc' -> pop
+    freq, freq_meta = merge_freq_arrays(
+        farrays=[ht.v2_exomes.freq, ht.v2_genomes.freq],
+        fmeta=[ht.index_globals().freq_meta, ht.index_globals().freq_meta_1],
+    )
+
+    # Select to first 28 groups, containing:
+    # Raw, adj, adj male & female, and adj sex-split pops for 8 pops.
+    # Hail sets this to first 28 of each entry in the array, not first 28 of array.
+    freq = freq[:28]
+    freq_meta = freq_meta[:28]
+
+    # Create a new struct to hold all joint information.
+    ht = ht.annotate(v2_joint=hl.struct(joint_freq=freq))
+
+    ht = ht.annotate_globals(
+        joint_freq_meta=freq_meta,
+        joint_freq_index_dict=make_freq_index_dict_from_meta(hl.literal(freq_meta)),
+    )
+
+    # Checkpoint output to new temp file.
+    ht = ht.checkpoint(hl.utils.new_temp_file("three_dup_genes_liftover", "ht"))
+
+    # Compute FAF on the merged exomes + genomes frequencies.
+    logger.info("Generating faf metrics...")
+    faf, faf_meta = faf_expr(
+        ht.v2_joint.joint_freq,
+        ht.joint_freq_meta,
+        ht.locus,
+        pops_to_exclude=POPS_TO_REMOVE_FOR_POPMAX,
+        pop_label="pop",
+    )
+    faf_meta_by_pop = {
+        m.get("pop"): i for i, m in enumerate(faf_meta) if m.get("pop") and len(m) == 2
+    }
+    faf_meta_by_pop = hl.literal(faf_meta_by_pop)
```

Comment:
what's the purpose of these lines?

Reply:
this comment hasn't been addressed

Reply:
ah, this was used when annotating grpmax with a faf95 field, but this can be dropped since we're not using it